charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -179,7 +179,7 @@ spec:
         emptyDir: { }
       {{- end }}
       {{- if .Values.global.tls.enabled }}
-      {{- if not (and .Values.externalServers.enabled .Values.externalServers.useSystemRoots) }}
+      {{- if or (not (and .Values.externalServers.enabled .Values.externalServers.useSystemRoots)) .Values.client.enabled }}
```

Comment:
If I'm reading this logic correctly, this will add the `consul-ca-cert` volume when clients are enabled regardless of whether  `useSystemRoots` is set. I don't think we want to do that. When `useSystemRoots` is true, you probably haven't provided the CA cert secret that this volume needs, and your deployment will fail.

Reply:
So just to make sure of the behavior then @ishustava -- when you have `client.enabled` and are using TLS, but also using system roots with external servers, is it possible to configure helm to use anything other than auto-encrypt?

For example, if someone was using something like HCP could they bring their own root that they want the client node certificates signed with? Because that would seem to be the case that this would cover -- making sure that whatever the root used in configuring the client nodes was able to be mounted to establish connections to clients. If not, it sounds like we only ever support auto-encrypt for client nodes when using publicly trusted certs for external servers?

Reply:
> For example, if someone was using something like HCP could they bring their own root that they want the client node certificates signed with?

This is not something we support with HCP. HCP only supports auto-encrypt. Plus, we provide a separate CA cert for RPC comms that client should be using to talk to servers. And so in the case of HCP, you would provide the secret and the volume will be mounted (although it won't be used for the api gw deployment because there's nothing here that uses RPC APIs). 

I'm thinking of a case when you created a cert for the servers that is signed by a well-known CA. In that case, you probably don't have access to the CA key, and so you can only use auto-encrypt for client certs. You would then configure your helm chart to use external servers with `useSystemRoots` and auto-encrypt enabled. I can't think of a use case when someone would need to use `useSystemRoots` without auto-encrypt otherwise.

Reply:
Got it, that makes sense here then -- we'll still need some variation of the guard above for actually mounting the auto-encrypt cert regardless of `useSystemRoots` when clients are enabled in the actual controller container, but given that this case: `useSystemRoots` + clients + no auto-encrypt isn't actually something we support, I'm thinking that as a follow-up to this we should explicitly disallow that configuration.

I'll try and push the recommended changes here to drop the "bring your own cert" case.

Reply:
Thank you, @andrewstucki. Feel free to take control of this branch.